pypreprocess/reporting/pypreproc_reporter.py

```diff
@@ -0,0 +1,1001 @@
+
+    """ 
+    Creates plots associated with corregistration 
+    and returns them as SVG url embedded in HTML.
+    Calls generate_registration_plot. 
+    
+    Parameters
+    ----------
+     subject_data: `SubjectData` instance
+       object that encapsulates the date for the subject (should have fields
+       like func, anat, output_dir, etc.)
+
+    output_dir: string
+        directory containing all the output files
+    
+    report_path: string, optional (default None)
+        path to the report HTML file
+
+    """
+
+    subject_data._set_session_ids()
+
+    if subject_data.anat is None:
+        print("Subject 'anat' field is None; nothing to do")
+        return
+
+    src, ref = subject_data.func, subject_data.anat
+    src_brain, ref_brain = "mean_functional_image", "anatomical_image"
+    if not coreg_func_to_anat:
+        src, ref = ref, src
+        src_brain, ref_brain = ref_brain, src_brain
+
+    heading = "Corregistration %s == > %s" % (src_brain, ref_brain)
+
+    for_substitution = {}
+ 
+    if log:
+        for_substitution['heading'] = heading
+        for_substitution['id_link'] = heading.replace(" ", "_")
+        for_substitution['log'] = get_log_text(src)
+        for_substitution['log_link'] = 'file:///'+os.path.join(output_dir,
+                                        'nilearn_report_log.html#'
+                                        )+for_substitution['id_link']
+        log_text = embed_in_HTML('log_sub_template.html',
+                                for_substitution)
+    else:
+        log_text = None
+
+    return generate_registration_report((ref, ref_brain),
+            (src, src_brain), output_dir, for_substitution,
+            report_path=report_path), log_text
+
+
+def generate_segmentation_report(subject_data, output_dir,
+    subject_gm_file=None, subject_wm_file=None, subject_csf_file=None,
+    comment="", only_native=False, tooltip=None,
+    log=True, report_path=None):
+    
+    """ 
+    Creates plots associated with segmentation 
+    and returns them as SVG url embedded in HTML. 
+    
+    Parameters
+    ----------
+     subject_data: `SubjectData` instance
+       object that encapsulates the date for the subject (should have fields
+       like func, anat, output_dir, etc.)
+
+    output_dir: string
+        directory containing all the output files
+    
+    report_path: string, optional (default None)
+        path to the report HTML file
+
+    """
+    _set_templates()
+
+    seg_plot = []
+    for_substitution = {}
+
+    subject_data._set_session_ids()
+
+    segmented = False
+    for item in ['gm', 'wm', 'csf']:
+        if hasattr(subject_data, item):
+            segmented = True
+            break
+    if not segmented:
+        return
+
+    if log:
+        for_substitution['heading'] = 'Segmentation'
+        for_substitution['id_link'] = for_substitution['heading']
+        for_substitution['log'] = get_log_text(getattr(subject_data,
```

Comment:
```suggestion
        for_substitution['log'] = get_log_text(
            getattr(subject_data, 'gm') or
            getattr(subject_data, 'csf') or...
```
would be more readable.